nilearn/tests/test_surface.py

```diff
@@ -0,0 +1,393 @@
+    # proj matrix has shape (n_vertices, img_size)
+    assert_equal(proj.shape, (5 * 7, 5 * 7 * 13))
+    # proj.dot(img) should give the values of img at the vertices' locations
+    values = proj.dot(img.ravel()).reshape((5, 7))
+    assert_array_almost_equal(values, img[:, :, 0])
+    mesh = _flat_mesh(5, 7)
+    proj = surface.projection_matrix(
+        mesh, np.eye(4), (5, 7, 1), radius=.1, n_points=10)
+    assert_array_almost_equal(proj.toarray(), np.eye(proj.shape[0]))
+    mask = np.ones(img.shape, dtype=int)
+    mask[0] = 0
+    proj = surface.projection_matrix(
+        mesh, np.eye(4), img.shape, radius=2., n_points=10, mask=mask)
+    proj = proj.toarray()
+    # first row of the mesh is masked
+    assert_array_almost_equal(proj.sum(axis=1)[:7], np.zeros(7))
+    assert_array_almost_equal(proj.sum(axis=1)[7:], np.ones(proj.shape[0] - 7))
+    # mask and img should have the same shape
+    assert_raises(ValueError, surface.projection_matrix,
+                  mesh, np.eye(4), img.shape, mask=np.ones((3, 3, 2)))
+
+
+def test_sampling_affine():
+    # check sampled (projected) values on a toy image
+    img = np.ones((4, 4, 4))
+    img[1, :, :] = 2
+    nodes = [[1, 1, 2], [10, 10, 20], [30, 30, 30]]
+    mesh = [np.asarray(nodes), None]
+    affine = 10 * np.eye(4)
+    affine[-1, -1] = 1
+    texture = surface._nearest_voxel_sampling(
+        [img], mesh, affine=affine, radius=1, kind='ball')
+    assert_array_equal(texture[0], [1., 2., 1.])
+    if LooseVersion(scipy.__version__) < LooseVersion('0.14'):
+        return
+    texture = surface._interpolation_sampling(
+        [img], mesh, affine=affine, radius=0, kind='ball')
+    assert_array_almost_equal(texture[0], [1.1, 2., 1.])
+
+
+def test_sampling():
+    mesh = _flat_mesh(5, 7, 4)
+    img = _z_const_img(5, 7, 13)
+    mask = np.ones(img.shape, dtype=int)
+    mask[0] = 0
+    projectors = [surface._nearest_voxel_sampling]
+    if LooseVersion(scipy.__version__) >= LooseVersion('0.14'):
+        projectors.append(surface._interpolation_sampling)
+    for kind in ('line', 'ball'):
+        for projector in projectors:
+            projection = projector([img], mesh, np.eye(4),
+                                   kind=kind, radius=0.)
+            assert_array_almost_equal(projection.ravel(), img[:, :, 0].ravel())
+            projection = projector([img], mesh, np.eye(4),
+                                   kind=kind, radius=0., mask=mask)
+            assert_array_almost_equal(projection.ravel()[7:],
+                                      img[1:, :, 0].ravel())
+            assert_true(np.isnan(projection.ravel()[:7]).all())
+
+
+def test_vol_to_surf():
+    # test 3d niimg to cortical surface projection and invariance to a change
+    # of affine
+    mni = datasets.load_mni152_template()
+    mesh = _generate_surf()
+    _check_vol_to_surf_results(mni, mesh)
+    if LooseVersion(nb.__version__) > LooseVersion('1.2.0'):
```

Comment:
We don't need this. We dropped supporting nibabel=1.2.0. Minimum version is 2.0.2